### PR TITLE
[Temp PR] SinkParser 

### PR DESCRIPTION
**Pull request by minsunny**
https://nanopay.atlassian.net/browse/CPF-2244

will be gradually extended one by one here

### CODE REVIEW

File: src/foam/u2/Element.js [hunk 1]
```diff
@@ -2648,6 +2648,17 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'digColumns',
```
Comment (mcarcaso) [inline src/foam/u2/Element.js]: digColumns has nothing to do with u2. Undo these changes. We'll introduce digColumns to models when we have everything else working.

File: src/foam/nanos/dig/DIG.js [hunk 2]
```diff
@@ -53,6 +53,18 @@ foam.CLASS({
     },
     'cmd',
     'format',
+    {
+      name: 'digColumns',
```
Comment (mcarcaso) [inline src/foam/nanos/dig/DIG.js]: Make this a `class: 'StringArray'`. Doing so means you won't be able to use an expression unless you do a
`factory: null`

File: src/foam/nanos/dig/DIG.js [hunk 3]
```diff
@@ -53,6 +53,18 @@ foam.CLASS({
...
+      expression: function(daoKey) {
+         var dao = this.__context__[daoKey];
```
Comment (mcarcaso) [inline src/foam/nanos/dig/DIG.js:61]: You might want to ensure that `dao` isn't null before attempting to use it.

File: src/foam/nanos/dig/DIG.js [hunk 4]
```diff
@@ -53,6 +53,18 @@ foam.CLASS({
...
+
+         if ( dao.of.getAxiomByName('digColumns') ) {
+          return dao.of.getAxiomByName('digColumns').columns.toString();
```
Comment (mcarcaso) [inline src/foam/nanos/dig/DIG.js]: Remove this for now. We'll introduce the `digColumns` axiom when we're more ready later.

File: src/foam/nanos/dig/DIG.js [hunk 5]
```diff
@@ -53,6 +53,18 @@ foam.CLASS({
...
+        } else {
+          return dao.of.getAxiomsByClass(foam.core.Property).toString();
```
Comment (mcarcaso) [inline src/foam/nanos/dig/DIG.js]: this isn't right. You want an array of strings, not the string of an array.
`return dao.of.getAxiomsByClass(foam.core.Property).map(p => p.name);`

File: src/foam/nanos/dig/DIG.js (hunk 2 above)
Comment (mcarcaso) [inline src/foam/nanos/dig/DIG.js]: I'd rename this to either `columns` or `fields`
I like `fields` better because they're not always rendered as columns, as is the case for a JSON.

File: src/foam/nanos/dig/DIG.js [hunk 6]
```diff
@@ -53,6 +53,16 @@ foam.CLASS({
     },
     'cmd',
     'format',
+    {
+      class: 'StringArray',
+      name: 'fields',
+      expression: function(daoKey) {
+         var dao = this.__context__[daoKey];
+
+         if ( dao != null )
```
Comment (mcarcaso) [inline src/foam/nanos/dig/DIG.js]: You do not need this if statement now that you check if dao is null in the return statement.

File: src/foam/nanos/dig/DIG.js [hunk 7]
```diff
@@ -138,6 +148,11 @@ foam.CLASS({
           query = true;
           url += "q=" + encodeURIComponent(q);
         }
+        if ( fields ) {
```
Comment (mcarcaso) [inline src/foam/nanos/dig/DIG.js]: fields will always be an array so this check will never be false. Maybe you want to check if fields.length > 0

**Comment by minsunny** [comment]
close this and moved here https://github.com/foam-framework/foam2/pull/2480
